ssl/quic/quic_port.c

```diff
@@ -1090,12 +1078,12 @@ static int port_validate_token(QUIC_PKT_HDR *hdr, QUIC_PORT *port,
          */
         if (!memcmp(&token.rscid, &hdr->dst_conn_id, sizeof(QUIC_CONN_ID)))
```

Comment:
This is the source of you memory santizer CI error (at least the one that I can see).  You can't do a memcmp over the entire sizeof(QUIC_CONN_ID), because the connection id may not use that entire 20 byte array in that struct.  You correctly only copy in rscid.id_len in the marshalling code, but you also need to only check that amount here.  The check should probably be something like:
```
if (rscid.id_len != hdr.dst_conn_id.id_len || !memcmp(rscid.id, &hdr.dst_conn_id.id, rscid.id_len))
    return 0;
```

That may also fix up the other CI errors, in the event that you erroenously failing this check by comparing random values at the end of that array.